### ISSUE TITLE
Wrong "unbound" error when declaring class in "if" block

### ISSUE THREAD

**Issue by giyyapan** [microsoft/pyright#199]
**Describe the bug**
Please see the screenshot.
![image](https://user-images.githubusercontent.com/3479453/63009642-79025780-beb7-11e9-9299-3b53436fefe9.png)

This also happens to some other block (while, for), but ok in function block

![image](https://user-images.githubusercontent.com/3479453/63009852-e57d5680-beb7-11e9-8960-53a5070e8555.png)

**To Reproduce**
Full code:
```python
if True:
    class X:
        pass

    a: X # <- this works

    class A:
        a: X # <- unbound error
        b = X # <- unbound error

        def fn(self) -> X:... # <- unbound error
```

**Expected behavior**

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```
def foo(self) -> str:
    return 3
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool? Which version? You can find the version of the VS Code extension by clicking on the Pyright icon in the extensions panel.

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. I've tracked down the cause of this bug, and it will be fixed in the next published version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.56, which I just published.
